PWGHF/HFC/TableProducer/correlatorDstarHadrons.cxx

```diff
@@ -0,0 +1,285 @@
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file correlatorDstarHadron.cxx
+/// \author Deependra Sharma <deependra.sharma@cern.ch>, IITB
+/// \author Fabrizio Grosa <fabrizio.grosa@cern.ch>, CERN
+
+// O2
+#include "CommonConstants/PhysicsConstants.h"
+#include "Framework/ASoAHelpers.h"
+#include "Framework/AnalysisTask.h"
+#include "Framework/runDataProcessing.h"
+
+// O2Physics
+#include "Common/DataModel/Multiplicity.h"
+
+// PWGHF
+#include "PWGHF/HFC/DataModel/CorrelationTables.h"
+#include "PWGHF/DataModel/CandidateReconstructionTables.h"
+#include "PWGHF/DataModel/CandidateSelectionTables.h"
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+
+// flaging a collision if D* meson is found.
+struct HfCollisionSelector {
+  Produces<aod::DmesonSelection> collisionWDstar;
+
+  Configurable<bool> selectionFlagDstar{"selectionFlagDstar", true, "selection flag for Dstar"};
+  Configurable<float> yCandMax{"yCandMax", 0.8, "max. cand. rapidity"};
+  Configurable<float> ptCandMin{"ptCandMin", 1., "min. cand. pT"};
+
+  using DstarCandidates = soa::Join<aod::HfCandDstar, aod::HfSelDstarToD0Pi>;
+  using FilteredCandidates = soa::Filtered<DstarCandidates>;
+
+  SliceCache cache;
+  Preslice<DstarCandidates> perColDstarCand = aod::hf_cand::collisionId;
+
+  // candidates who passed the slection criteria defined in "CandidateSelectionTables.h"
+  Filter candidateFilter = aod::hf_sel_candidate_dstar::isSelDstarToD0Pi == selectionFlagDstar;
+
+  void processCollisionSelWDstar(aod::Collisions const& collisions,
+                                 FilteredCandidates const& candidates)
+  {
+
+    for (const auto& collision : collisions) {
+      bool isDstarFound = false;
+      auto candidatesPerCol = candidates.sliceByCached(aod::hf_cand::collisionId, collision.globalIndex(), cache);
+      if (!(candidatesPerCol.size() > 0)) {
+        collisionWDstar(isDstarFound); // compatible with collision table (filled collision by collision)
+        continue;
+      }
+      for (const auto& candidate : candidatesPerCol) {
+        auto yDstar = candidate.y(constants::physics::MassDStar);
+        auto ptDstar = candidate.pt();
+        if (yCandMax >= 0 && std::abs(yDstar) > yCandMax) {
+          continue;
+        }
+        if (ptCandMin >= 0 && ptDstar < ptCandMin) {
+          continue;
+        }
+        isDstarFound = true;
+        break;
+      } // candidate loop
+      LOG(info) << "processCollisionSelWDstar: isDstarFound = " << isDstarFound;
+      collisionWDstar(isDstarFound); // compatible with collision table (filled collision by collision)
+    }                                // collision loop
+  }
+  PROCESS_SWITCH(HfCollisionSelector, processCollisionSelWDstar, "process only data for dstar hadron correlation", true);
```

Comment:
Why do you need a special name and a switch for a single process function?

Reply:
Right Now I am adding process function only for Data. In future I have to add process function for McRec and McGen also. I thought to do that in next PR.